src/library/dao/cuedao.cpp

```diff
@@ -53,9 +53,16 @@ CuePointer CueDAO::cueFromRow(const QSqlQuery& query) const {
     int length = record.value(record.indexOf("length")).toInt();
     int hotcue = record.value(record.indexOf("hotcue")).toInt();
     QString label = record.value(record.indexOf("label")).toString();
-    int iColorId = record.value(record.indexOf("color")).toInt();
-    PredefinedColorPointer color = Color::kPredefinedColorsSet.predefinedColorFromId(iColorId);
-    CuePointer pCue(new Cue(id, trackId, (Cue::Type)type, position, length, hotcue, label, color));
+    uint colorValue = record.value(record.indexOf("color")).toUInt();
```

Comment:
RgbColor::fromQVariant

Reply:
`fromQVariant` returns `RgbColor::optional_t`, not plain `RgbColor`.

Reply:
Correct, you need to handle the result and map it to a non-optional RgbColor. But no reason to implement a redundant conversion from QVariant to RgbColor -> violates DRY